pkg/openstack/loadbalancer.go

```diff
@@ -1340,6 +1340,37 @@ func (lbaas *LbaasV2) ensureOctaviaPool(lbID string, listener *listeners.Listene
 	if err != nil && err != openstackutil.ErrNotFound {
 		return nil, fmt.Errorf("error getting pool for listener %s: %v", listener.ID, err)
 	}
+
+	// By default, use the protocol of the listener
+	poolProto := v2pools.Protocol(listener.Protocol)
+	if svcConf.enableProxyProtocol {
+		poolProto = v2pools.ProtocolPROXY
+	} else if (svcConf.keepClientIP || svcConf.tlsContainerRef != "") && poolProto != v2pools.ProtocolHTTP {
+		poolProto = v2pools.ProtocolHTTP
+	}
+
+	// Delete the pool and its members if it already exists and has the wrong protocol
+	if pool != nil && v2pools.Protocol(pool.Protocol) != poolProto {
+		klog.V(2).Infof("Deleting wrong pool %s for listener %s because the protocol does not match", pool.ID, listener.ID)
+
+		// Delete pool automatically deletes all its members.
+		mc := metrics.NewMetricContext("loadbalancer_pool", "delete")
+		if err = v2pools.Delete(lbaas.lb, pool.ID).ExtractErr(); err != nil {
```

Comment:
`if err = v2pools.Delete(lbaas.lb, pool.ID).ExtractErr(); mc.ObserveRequest(err) != nil {` so you could remove Line 1366

Reply:
But this would also record `IsNotFound` as an error in the metrics, would this be correct? We are doing the same in line 2383.

Reply:
Hmm, yes, you are right. I didn't notice that.